Architecture.md

```diff
@@ -1,32 +1,46 @@
 # Architecture
 
-The general flow of data in the library is as follows:
-1. User creates `FileData` object ([code link](https://github.com/realm/realm-dotnet-lfs/blob/630b323b31e45a25fee9fac4bf745c8b9123c34a/Realm.LFS/FileData.cs#L57))
-2. We store the Stream to a file on disk ([code link](https://github.com/realm/realm-dotnet-lfs/blob/630b323b31e45a25fee9fac4bf745c8b9123c34a/Realm.LFS/FileData.cs#L59))
-3. We try to upload the file (with incremental backoffs on failure) ([code link](https://github.com/realm/realm-dotnet-lfs/blob/630b323b31e45a25fee9fac4bf745c8b9123c34a/Realm.LFS/FileData.cs#L70))
-4. We update the `FileData` with the url of the newly uploaded file ([code link](https://github.com/realm/realm-dotnet-lfs/blob/630b323b31e45a25fee9fac4bf745c8b9123c34a/Realm.LFS/Managers/RemoteFileManager.cs#L143-L156))
+![Architecture diagram](https://github.com/realm/realm-dotnet-lfs/assets/2315687/fb0ade9d-a0f0-4f73-952e-ad7917e16a9d)
+
+The diagram above roughly describes the functionality of the library, but the general flow is as follows:
+
+1. User creates `FileData` object ([code link](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/FileData.cs#L70))
+2. We store the Stream to a file on disk ([code link](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/FileData.cs#L72))
+3. We try to upload the file (with incremental backoffs on failure) ([code link](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/FileData.cs#L83))
+4. We update the `FileData` with the url of the newly uploaded file ([code link](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/Managers/RemoteStorageManager.cs#L143-L156))
```

Comment:
This link is not working, but I suppose it should once it's all merged to main. Probably we should give a final "click review" for the links once it's merged 😁